nanoFramework.Device.Bluetooth/Security/DevicePairing.cs

```diff
@@ -0,0 +1,306 @@
+        /// </summary>
+        /// <returns></returns>
+        public DevicePairingResult Pair()
+        {
+            return Pair(_protectionLevel);
+        }
+
+        /// <summary>
+        /// Attempts to pair the device using a provided level of protection.
+        /// </summary>
+        /// <param name="minProtectionLevel">The required level of protection to use for the pairing action.</param>
+        /// <returns>The result of the pairing action.</returns>
+        public DevicePairingResult Pair(DevicePairingProtectionLevel minProtectionLevel)
+        {
+            _isPaired = false;
+
+            if (_server != null)
+            {
+                _server.Start();
+            }
+            else
+            {
+                _device.ConnectDeviceIfNotConnected();
+            }
+
+            SetPairAttributes();
+            
+            if (NativeStartPair(_device.ConnectionHandle) == 0)
+            { 
+                if (!_completedEvent.WaitOne(operationPairTimeout, false))
+                {
+                    _pairingStatus = DevicePairingResultStatus.AuthenticationTimeout;
+                }
+            }
+
+            return new DevicePairingResult(minProtectionLevel, _pairingStatus);
+        }
+
+        /// <summary>
+        /// Unpair the device.
+        /// </summary>
+        /// <returns>The result of the unpairing action.</returns>
+        public DeviceUnpairingResult Unpair()
+        {
+            return new DeviceUnpairingResult(0);
+        }
+
+        /// <summary>
+        ///  Gets a value that indicates whether the device can be paired.
+        ///  **True** if the device can be paired, otherwise **false**.
+        /// </summary>
+        public bool CanPair { get => _canPair; }
+
+        /// <summary>
+        /// Gets a value that indicates whether the device is currently paired.
+        /// **True** if the device is currently paired, otherwise **false**.
+        /// </summary>
+        public bool IsPaired { get => _isPaired; }
+
+        /// <summary>
+        /// Returns if latest pairing was Authenicated.
+        /// </summary>
+        public bool IsAuthenticated { get => _isAuthenticated; }
+
+        /// <summary>
+        /// Set Pairing Attibutes in Native level
+        /// </summary>
+        internal void SetPairAttributes()
+        {
+            NativeSetPairAttributes();
+        }
+
+        /// <summary>
+        /// Internal event handler
+        /// </summary>
+        /// <param name="btEvent"></param>
+        internal void OnEvent(BluetoothEventSesssion btEvent)
+        {
+            switch (btEvent.type)
+            {
+                case BluetoothEventType.PassKeyActions:
+                    DevicePairingKinds kind = (DevicePairingKinds)btEvent.data;
+
+                    // Debug.WriteLine($"# Pairing DevicePairingKinds:{kind} status:{btEvent.status}");
+
+                    OnEvent(btEvent, kind, 0);
+
+                    //    // None
```

Comment:
You may want to use the 4 slash instead or double plus spaces plus double.

Reply:
No need for double slashes. Made it normal comments.